syntax = "proto3";

option go_package = "github.com/thiagorobert/space-api/autogenerated_proto";

package space_api_proto;

import "google/api/annotations.proto";

service Tle {
  rpc Decode(TleDecodeReq) returns (TleDecodeRes) {
    option (google.api.http) = {
      post: "/space/v1/tle/decode"
      body: "*"
    };
  }
  rpc ToOrbit(TleToOrbitReq) returns (TleToOrbitRes) {
    option (google.api.http) = {
      post: "/space/v1/tle/orbit"
      body: "*"
    };
  }
  rpc ToCorridor(TleToCorridorReq) returns (TleToCorridorRes) {
    option (google.api.http) = {
      post: "/space/v1/tle/corridor"
      body: "*"
    };
  }
}

message TleData {
  string name = 1;
  string line1 = 2;
  string line2 = 3;
}

message TleDecodeReq {
  TleData tle_data = 1;
}

message TleDecodeRes {
  string decoded = 1;
}

message TleToOrbitReq {
  TleData tle_data = 1;
}

message TleToOrbitRes {
  string orbit = 1;
  string visualization = 2;
}

message TleToCorridorReq {
  TleData tle_data = 1;
}

message TleToCorridorRes {
  string corridor = 1;
}
